### PR TITLE
CATROID-569 REFACTOR DeleteItemOfUserListAction to Kotlin

### PR DESCRIPTION
**Pull request by Rawbat**
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    private var formulaIndexToDelete: Formula? = null
+    private var userList: UserList? = null
+
+    override fun update(percent: Float) {
+        if (userList == null) {
+            return
+        }
+        if (userList!!.value.size == 0) {
+            return
+        }
```
Comment (mandelbrot2357) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt:44]:
```suggestion
        if (userList?.value?.size ?: 0 == 0) {
            return
        }
```

Null safety is Kotlin's biggest strength. Java style checks defeat its purpose. Also see below.

File: catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            } else {
+                formulaIndexToDelete!!.interpretInteger(sprite)
+            }
+        } catch (interpretationException: InterpretationException) {
+            Log.e(DeleteItemOfUserListAction::class.java.simpleName,
+                "Interpreting sprite as integer failed : " + sprite.toString(), interpretationException)
+            1
+        }
+
+        indexToDelete--
```
Comment (mandelbrot2357) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt:58]:
```suggestion
        val indexToDelete = try {
            formulaIndexToDelete?.interpretInteger(sprite)?.minus(1) ?: 0
        } catch (interpretationException: InterpretationException) {
            Log.e(
                DeleteItemOfUserListAction::class.java.simpleName,
                Interpreting sprite as integer failed : ${sprite.toString()}",
                interpretationException
            )
            0
        }
```

File: catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+        if (indexToDelete >= userList!!.value.size || indexToDelete < 0) {
+            return
+        }
+
+        (userList!!.value as ArrayList<Object?>).removeAt(indexToDelete)
```
Comment (mandelbrot2357) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt:64]:
```suggestion
        if (indexToDelete in 0..userList!!.value.size) {
            userList!!.value.removeAt(indexToDelete)
        }
```

File: catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        this.userList = userVariable
+    }
+
+    fun setFormulaIndexToDelete(formulaIndexToDelete: Formula?) {
+        this.formulaIndexToDelete = formulaIndexToDelete
+    }
+
+    fun setSprite(sprite: Sprite?) {
+        this.sprite = sprite
+    }
```
Comment (mandelbrot2357) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt:77]:
```suggestion
```

You can delete all of these. Kotlin will generate them automatically based on member visibility. Also, setters should not accept null.

File: catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.Formula
+import org.catrobat.catroid.formulaeditor.InterpretationException
+import org.catrobat.catroid.formulaeditor.UserList
+import java.util.ArrayList
+
+class DeleteItemOfUserListAction : TemporalAction() {
+    private var sprite: Sprite? = null
+    private var formulaIndexToDelete: Formula? = null
+    private var userList: UserList? = null
```
Comment (mandelbrot2357) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/DeleteItemOfUserListAction.kt:36]:
```suggestion
    var sprite: Sprite? = null
    var formulaIndexToDelete: Formula? = null
    var userList: UserList? = null
```

See setters below

**Comment by christophsiegl** [comment]
Thanks - looks nice! Also a great review!

**Pull request merged**